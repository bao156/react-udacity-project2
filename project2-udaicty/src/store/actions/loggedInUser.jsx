export const ADD_LOGGED_IN_USER = "ADD_LOGGED_IN_USER";
export const AUTHENTICATE_USER = "AUTHENTICATE_USER";

function addLoggedInUser(loggedInUser) {
  return {
    type: ADD_LOGGED_IN_USER,
    payload: loggedInUser,
  };
}

export function handleAddingLoggedInUser(user) {
  return (dispatch) => {
    dispatch(addLoggedInUser(user));
  };
}
