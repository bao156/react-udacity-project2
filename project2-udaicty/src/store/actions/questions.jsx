import { _getQuestions } from "../../_DATA";

export const ADD_LIST_OF_QUESTIONS = "ADD_LIST_OF_QUESTIONS";

function addListOfQuestions(questions) {
  return {
    type: ADD_LIST_OF_QUESTIONS,
    questions,
  };
}

export function fetchAllQuestions() {
  return (dispatch) => {
    _getQuestions().then((questions) => {
      dispatch(addListOfQuestions(Object.values(questions)));
    });
  };
}
