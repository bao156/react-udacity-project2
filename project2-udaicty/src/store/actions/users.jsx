import { _getUsers } from "../../_DATA";

export const ADD_LIST_OF_USERS = "ADD_LIST_OF_USERS";

function addListOfUsers(users) {
  return {
    type: ADD_LIST_OF_USERS,
    users,
  };
}

export function fetchAllUser() {
  return (dispatch) => {
    _getUsers().then((users) => {
      dispatch(addListOfUsers(users));
    });
  };
}
