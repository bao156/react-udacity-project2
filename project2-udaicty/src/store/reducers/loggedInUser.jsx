import { ADD_LOGGED_IN_USER } from "../actions/loggedInUser";

export default function loggedInUser(state = "", action) {
  switch (action.type) {
    case ADD_LOGGED_IN_USER:
      return (state = action.payload);
    default:
      return state;
  }
}
