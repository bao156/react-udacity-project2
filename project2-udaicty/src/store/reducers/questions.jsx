import { ADD_LIST_OF_QUESTIONS } from "../actions/questions";

export default function questions(state = "", action) {
  switch (action.type) {
    case ADD_LIST_OF_QUESTIONS:
      return (state = action.questions);
    default:
      return state;
  }
}
