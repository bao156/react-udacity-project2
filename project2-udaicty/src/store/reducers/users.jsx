import { ADD_LIST_OF_USERS } from "../actions/users";

export default function users(state = "", action) {
  switch (action.type) {
    case ADD_LIST_OF_USERS:
      return (state = action.users);
    default:
      return state;
  }
}
