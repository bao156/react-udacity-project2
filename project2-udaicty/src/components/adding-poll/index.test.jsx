import * as React from "react";
import { fireEvent, render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import loggedInUser from "../../store/reducers/loggedInUser";
import users from "../../store/reducers/users";
import questions from "../../store/reducers/questions";
import { BrowserRouter } from "react-router-dom";
import AddPollComponent from ".";

const store = configureStore({
  reducer: {
    loggedInUser: loggedInUser,
    users: users,
    questions: questions,
  },
  preloadedState: {
    loggedInUser: {
      payload: {
        id: "sarahedo",
        password: "password123",
        name: "Sarah Edo",
        avatarURL:
          "https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80",
        answers: {
          "8xf0y6ziyjabvozdd253nd": "optionOne",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
          am8ehyc8byjqgar0jgpub9: "optionTwo",
          loxhs1bqm25b708cmbf3g: "optionTwo",
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
      },
    },
    users: {
      users: {
        sarahedo: {
          id: "sarahedo",
          password: "password123",
          name: "Sarah Edo",
          avatarURL:
            "https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80",
          answers: {
            "8xf0y6ziyjabvozdd253nd": "optionOne",
            "6ni6ok3ym7mf1p33lnez": "optionOne",
            am8ehyc8byjqgar0jgpub9: "optionTwo",
            loxhs1bqm25b708cmbf3g: "optionTwo",
          },
          questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
        },
      },
    },
    questions: {
      questions: [
        {
          id: "8xf0y6ziyjabvozdd253nd",
          author: "sarahedo",
          timestamp: 1467166872634,
          optionOne: {
            votes: ["sarahedo"],
            text: "Build our new application with Javascript",
          },
          optionTwo: {
            votes: [],
            text: "Build our new application with Typescript",
          },
        },
        {
          id: "am8ehyc8byjqgar0jgpub9",
          author: "sarahedo",
          timestamp: 1488579767190,
          optionOne: {
            votes: [],
            text: "conduct a release retrospective 1 week after a release",
          },
          optionTwo: {
            votes: ["sarahedo"],
            text: "conduct release retrospectives quarterly",
          },
        },
      ],
    },
  },
});

describe("Adding poll component", () => {
  it("matches the snapshot when rendering", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddPollComponent />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it("will display a success message if all fields are submitted", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddPollComponent />
        </BrowserRouter>
      </Provider>
    );

    var option1Input = component.getByTestId("option1-input");
    fireEvent.change(option1Input, { target: { value: "option 1" } });
    var option2Input = component.getByTestId("option2-input");
    fireEvent.change(option2Input, { target: { value: "option 2" } });
    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.getByTestId("option1-input")).toBeInTheDocument();
  });
});
