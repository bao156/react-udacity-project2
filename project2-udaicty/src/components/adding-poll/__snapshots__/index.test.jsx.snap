// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding poll component matches the snapshot when rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="css-a988vp"
      >
        Would You Rather
      </p>
      <p
        class="css-9ev25s"
      >
        Create Your Own Poll
      </p>
      <div
        class="css-10jxxd2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1g7lsi9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-required"
            id="outlined-required-label"
          >
            First option
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="option1-input"
              id="outlined-required"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  First option
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1g7lsi9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-required"
            id="outlined-required-label"
          >
            Second Option
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="option2-input"
              id="outlined-required"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Second Option
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1h8o074-MuiButtonBase-root-MuiButton-root"
          data-testid="submit-button"
          tabindex="0"
          type="button"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="css-a988vp"
    >
      Would You Rather
    </p>
    <p
      class="css-9ev25s"
    >
      Create Your Own Poll
    </p>
    <div
      class="css-10jxxd2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1g7lsi9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-required"
          id="outlined-required-label"
        >
          First option
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="option1-input"
            id="outlined-required"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                First option
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1g7lsi9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-required"
          id="outlined-required-label"
        >
          Second Option
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="option2-input"
            id="outlined-required"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Second Option
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1h8o074-MuiButtonBase-root-MuiButton-root"
        data-testid="submit-button"
        tabindex="0"
        type="button"
      >
        Submit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
